e2e/my-posts.spec.ts

```diff
@@ -1,6 +1,27 @@
+import type { Page } from "@playwright/test";
 import test, { expect } from "@playwright/test";
 import { articleExcerpt, loggedInAsUserOne } from "./utils";
 
+async function openPublishedTab(page: Page) {
+  await page.goto("http://localhost:3000/my-posts");
+  await page.getByRole("link", { name: "Published" }).click();
+  await expect(page).toHaveURL(/\/my-posts\?tab=published$/);
```

Comment:
this might create flakiness if there is a delay in clicking the button and the url being updated with the published tab query. 

we can add something like 
`await page.waitForURL('url pattern');`
then assert the url

Reply:
Thank you for this, @pkspyder007; this is added, too.